xzes40/transformer/src/daemon.cpp

```diff
@@ -0,0 +1,144 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright 2017, the Oregon State University Capstone Project 'XZES40'
+// with the Apache Software Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///////////////////////////////////////////////////////////////////////////////
+
+
+///////////////////////////////////////////////////////////////////////////////
+// Filename: daemon.cpp
+// Description: Creattes daemon binary which waits and spawns jobs for
+//   transformations.
+// Mad props to:
+// - https://github.com/troydhanson/network/blob/master/unixdomain/srv.c
+///////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <stdlib.h>
+#include <csignal>
+
+#include <lib.hpp>
+#include <transform.hpp>
+#include <daemon.hpp>
+
+#define BUFFER_SIZE 1024
+
+// TODO: CHANGE THIS TO /var/run/... in the final release
+const char *socket_path = "/tmp/xzes40daemon.socket";
+
+int main(int argc, char* argv[]) {
+    int fd;
+
+    signal(SIGINT, xzes::signal_handler); 
+
+    if ((fd = xzes::setup_connection()) == -1)
+        exit(EXIT_FAILURE);
```

Comment:
what is the Unix socket connection? Does it wait for our server to make a connection request?